Assets/MixedRealityToolkit/Interfaces/Registrars/IMixedRealityServiceRegistrar.cs

```diff
@@ -0,0 +1,143 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.MixedReality.Toolkit.Core.Definitions.Utilities;
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.MixedReality.Toolkit.Core.Interfaces
+{
+    /// <summary>
+    /// Interface for Mixed Reality Toolkit service registration.
+    /// </summary>
+    public interface IMixedRealityServiceRegistrar
+    {
+        #region IMixedRealityService registration
+
+        /// <summary>
+        /// Registers a service of the specified type.
+        /// </summary>
+        /// <typeparam name="T">The interface type of the service to be registered (ex: IMixedRealityBoundarySystem).</typeparam>
+        /// <param name="serviceInstance">An instance of the service to be registered.</param>
+        bool RegisterService<T>(T serviceInstance) where T : IMixedRealityService;
```

Comment:
Would it be worth leaving a comment in remarks that calling RegisterService twice with the same service type will fail? 

Generally when I see Register/Unregister patterns, they are usually:

1) Register and you get back a handler, which you can use to unregister later.
2) Register and it must be one-per-thing you register, so unregister will either find that or not find that.

I think this design is option 2.

Reply:
Probably should call out what our return behavior is in this summary. I don't have a strong preference on whether or not a duplicate registration returns true or false. But we should be clear to users what the expected behavior is.

Reply:
agreed. one of the goals of this change was to limit behavioral change.

the team is definitely willing to entertain thoughts of how to make improvements to the existing locator, feel free to file issues and they can be addressed in the not too distant future